superset/db_engine_specs/doris.py

```diff
@@ -245,17 +248,45 @@ def adjust_engine_params(
         catalog: Optional[str] = None,
         schema: Optional[str] = None,
     ) -> tuple[URL, dict[str, Any]]:
-        database = uri.database
-        if schema and database:
-            schema = parse.quote(schema, safe="")
-            if "." in database:
-                database = database.split(".")[0] + "." + schema
-            else:
-                database = "internal." + schema
-            uri = uri.set(database=database)
-
+        if uri.database and "." in uri.database:
+            current_catalog, _ = uri.database.split(".", 1)
+        else:
+            current_catalog = "internal"
+
+        # In Apache Doris, each catalog has an information_schema for BI tool
+        # compatibility. See: https://github.com/apache/doris/pull/28919
+        adjusted_database = ".".join(
+            [catalog or current_catalog or "", "information_schema"]
+        ).rstrip(".")
```

Comment:
This may sound like a total nit, but I actually had some issues following what's going on here, especially the `catalog or current_catalog or ""` logic. As `current_catalog` is unnecessary if `catalog` is defined, I would have maybe just reused the latter variable for all these uses. Something like:

```python
        if catalog:
            pass
        elif uri.database and "." in uri.database:
            catalog, _ = uri.database.split(".", 1) or ""  # notice how I also moved the `or ""` part here
        else:
            catalog = "internal"
```
Then later just

```python
        adjusted_database = ".".join([catalog, "information_schema"])
```
Also, why is `.rstrip(".")` needed? I don't see how we can ever hit that, as `adusted_database` will always end with `.information_schema`, right?

Reply:
@villebro Yes, you are right. Thank you very much for your suggestion. I have modified it according to your suggestion.

Reply:
@villebro Thanks for your advice. After in-depth testing with Doris, we found that there is still a problem. The previous test only tested the case of linking data sources. When operating on SQL Lab, it will also go to this function and cannot use the information_schema library fixedly. When there is a schema value, the user-provided schema should be used for querying. This implementation is the correct behavior at present.
1. When linking data sources, the schema is empty and the information_schema library is used uniformly
2. When the schema has a value, the schema value provided by the user is used